scanner/services/indexer.go

```diff
@@ -104,14 +104,23 @@ func (s *indexerService) GetIndexReport(ctx context.Context, req *v4.GetIndexRep
 		"hash_id", req.GetHashId(),
 	)
 	zlog.Info(ctx).Msg("getting index report for container image")
+	ir, err := s.getIndexReport(ctx, req)
+	switch {
+	case errors.Is(err, errox.NotFound):
+		zlog.Warn(ctx).Err(err).Send()
+	case err != nil:
+		zlog.Error(ctx).Err(err).Msg("internal error: converting to v4.IndexReport")
+	}
+	return ir, err
+}
+
+func (s *indexerService) getIndexReport(ctx context.Context, req *v4.GetIndexReportRequest) (*v4.IndexReport, error) {
```

Comment:
Can we unwrap the `HashId` from the request outside the helper function so that we don't have to create a request object when calling from `GetOrCreateIndexReport()`?

Reply:
For sure. I did this quickly without thinking about it much so I missed room for improvement 😅